include/envoy/request_info/dynamic_metadata.cc

```diff
@@ -0,0 +1,9 @@
+#include "envoy/request_info/dynamic_metadata.h"
+
+namespace Envoy {
+namespace RequestInfo {
+
+std::atomic<size_t> DynamicMetadata::type_id_index_(0u);
```

Comment:
I think I'm a bit skeptical about state living in `include/envoy`. But, I don't see a good alternative, since I gather that other parts of the interface depend on this.

Reply:
Agreed on both counts.  Happy to brainstorm with you on this--I couldn't come up with a different way to do it with the templated interface.

Reply:
The other alternative is to put it in `source/common/request_info`. There is precedent here; we have various deps on `assert.h`, `protobuf.h` in `source/common/` already in `include/envoy`, but no `.cc`. I think it would be least bad to do this than add a `.cc` to `include/envoy`. But, this is bike shedding a bit. @mattklein123 ?

Reply:
I don't have a strong opinion. I guess what @htuch proposes is more inline w/ what we already have. I might drop a comment in here explaining why this is the way it is if we keep it this way since it's so unlike anything else we have.

Reply:
I have no feelings at all about how to implement this, so I moved the .cc file over to source/common/request_info and re-jiggered the dependencies.  I do want a review on that move before landing, LGTM or no :-}.

Reply:
Do you need to push? I don't see any changes.

Reply:
Nope, sorry, still working on other changes so haven't pushed yet.  We got into a bit of a bikeshed internally around one of Harvey's other comments.

Tangent question on github comment use:  I usually respond to these comments as I make changes in my local checkout, then flag that I'm ready for a new review after I push and the auto tests pass.  Is there a better pattern that would confuse reviewers less?  Or is it just that there was a long (not yet complete) gap between comment and push?

Reply:
There is no defined pattern, I just noticed that you said "I do want a review" and there was nothing to review. :) No problem.